internal/pkg/cli/pipeline_init.go

```diff
@@ -8,93 +8,244 @@ import (
 
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/ssm"
-	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
-	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
 
 	"github.com/spf13/cobra"
-	"github.com/spf13/afero"
+	"github.com/spf13/viper"
 )
 
+const (
+	pipelineAddEnvPrompt = "Would you like to add an environment to your pipeline?"
+	pipelineSelectEnvPrompt = "Which environment would you like to add to your pipeline?"
+	pipelineEnterGitHubRepoPrompt = "What is your application's GitHub repository?"
+
+	pipelineNoEnvError = "There were no more environments found that can be added to your pipeline. Please run `archer env init` to create a new environment."
+)
 
 // InitPipelineOpts holds the configuration needed to create a new pipeilne
 type InitPipelineOpts struct {
 	// Fields with matching flags.
-	GitHubRepo  string
+	Environments      []string
+	GitHubRepo        string
 	GitHubAccessToken string
-	Deploy bool
-	EnableCD bool
-	Environments []string
+	EnableCD          bool
+	Deploy            bool
 
-	// Interfaces to interact with dependencies.  // TODO
-	fs             afero.Fs
-	appStore       archer.ApplicationStore
-	manifestWriter archer.ManifestIO
-	prompt         prompter
+	// Interfaces to interact with dependencies.
+	envStore archer.EnvironmentStore
+	prompt   prompter
 
 	// Outputs stored on successful actions.
 	manifestPath string
 }
 
 // Ask prompts for fields that are required but not passed in.
 func (opts *InitPipelineOpts) Ask() error {
-	// TODO do we need?
-	// if opts.AppName == "" {
-	// 	if err := opts.askAppName(); err != nil {
-	// 		return err
-	// 	}
-	// }
-	// if opts.GitHubRepo == "" {
-	// 	if err := opts.askGitHubRepo(); err != nil {
-	// 		return err
-	// 	}
-	// }
-	// if opts.GitHubAccessToken == "" {
-	// 	if err := opts.askGitHubAccessToken(); err != nil {
-	// 		return err
-	// 	}
+	if len(opts.Environments) == 0 {
+		if err := opts.selectEnvironments(true); err != nil {
+			return err
+		}
+	}
+
+	if opts.GitHubRepo == "" {
+		if err := opts.selectGitHubRepo(); err != nil {
+			return err
+		}
+	}
+
+	if opts.GitHubAccessToken == "" {
+		if err := opts.getGitHubAccessToken(); err != nil {
+			return err
+		}
+	}
+
+	// if err := opts.askEnableCD(); err != nil {
+	// 	return err
 	// }
 
-	// if len(opts.Environments) == 0 {
-	// 	if err := opts.askEnvironments(); err != nil {
-	// 		return err
-	// 	}
+	// TODO ask this after pipeline.yml is written
+	// if err := opts.askDeploy(); err != nil {
+	// 	return err
 	// }
+
 	return nil
 }
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (opts *InitPipelineOpts) Validate() error {
 	// TODO
+	if viper.GetString(projectFlag) == "" {
```

Comment:
Instead of using `viper` directly, we changed the command structs to embed the `globalOpts` structs like here:
https://github.com/aws/amazon-ecs-cli-v2/blob/96599021f3209664893faf7db41718ac11119d87/internal/pkg/cli/app_package.go#L50

I think that should make testing easier :)